apps/typeform/frontend/src/Field/TypeFormField.tsx

```diff
@@ -0,0 +1,254 @@
+  ERROR = 'ERROR'
+}
+
+const initialState = {
+  error: false,
+  value: '',
+  selectedForm: {
+    name: '',
+    href: '',
+    isPublic: true,
+    id: ''
+  } as FormOption,
+  hasStaleData: false,
+  token: window.localStorage.getItem('token'),
+  forms: [] as FormOption[],
+  loading: true
+};
+
+const isStaleData = (value: string, forms: FormOption[]): boolean => {
+  if (value) {
+    if (forms.length === 0) {
+      return true;
+    } else {
+      if (forms.find(form => form.href === value)) {
+        return false;
+      } else {
+        return true;
+      }
+    }
+  } else {
+    return false;
+  }
+};
+
+const getSelectedForm = (value: string, forms: FormOption[]) => {
+  return forms.find(form => form.href === value) || initialState.selectedForm;
+};
+
+export function TypeFormField({ sdk }: Props) {
+  const { workspaceId } = sdk.parameters.installation as InstallationParameters;
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { loading, forms, value, hasStaleData, selectedForm, error, token } = state;
+  console.log(value);
+
+  function reducer(
+    state = initialState,
+    action: { type: string; payload?: any }
+  ): typeof initialState {
+    switch (action.type) {
+      case ACTION_TYPES.INIT: {
+        const { forms } = action.payload;
+        const currentFieldValue = sdk.field.getValue();
+        const hasStaleData = isStaleData(currentFieldValue, forms);
+        return {
+          ...state,
+          value: currentFieldValue,
+          selectedForm: getSelectedForm(currentFieldValue, forms),
+          loading: false,
+          forms,
+          error: false,
+          hasStaleData
+        };
+      }
+      case ACTION_TYPES.UPDATE_VALUE: {
+        const { value, forms } = action.payload;
+        let selectedForm = initialState.selectedForm;
+        if (value) {
+          sdk.field.setValue(value);
+          selectedForm = (forms as FormOption[]).find(form => form.href === value)!;
+        } else {
+          selectedForm = initialState.selectedForm;
+          sdk.field.removeValue();
+        }
+        return { ...state, value, selectedForm, hasStaleData: false };
+      }
+      case ACTION_TYPES.RESET: {
+        sdk.field.removeValue();
+        return {
+          ...state,
+          value: '',
+          hasStaleData: false,
+          selectedForm: initialState.selectedForm
+        };
+      }
+      case ACTION_TYPES.UPDATE_TOKEN: {
+        const { token } = action.payload;
+        return {
+          ...state,
+          loading: true,
+          error: false,
+          token
+        };
+      }
+      case ACTION_TYPES.ERROR: {
+        return { ...state, loading: false, error: true };
+      }
+      default:
+        return state;
+    }
+  }
+
+  useEffect(() => {
+    const fetchForms = async () => {
+      try {
+        const response = (await (
+          await fetch(`${process.env.LAMBDA_ENDPOINT}/forms/${workspaceId}/${token}`)
```

Comment:
Same question about why we don't have the token go into the header instead.

Reply:
Eliminate `process.env.LAMBDA_ENDPOINT`. It's not needed. Frontend and backend are served from the same domain, you can just use absolute URLs (`/forms/workspaceId`)